File: java/src/trie/randomTestData.java
package trie;

import java.util.Random;

class randomTestData {
	private static final Random rand = new Random(0);

	String[] strings;
	boolean hasPrefix;

	randomTestData() {
		var n = 1 + rand.nextInt(26);
		var m = 1 + rand.nextInt(20);

		strings = new String[1 + rand.nextInt(20)];
		for (var i = 0; i < strings.length; i++) {
			var buf = new char[1 + rand.nextInt(m)];
			for (var j = 0; j < buf.length; j++) {
				buf[j] = (char)('a' + rand.nextInt(n));
			}
			strings[i] = new String(buf);
		}

		for (var i = 0; i < strings.length; i++) {
			for (var j = 0; j < strings.length; j++) {
				if (i != j && strings[i].startsWith(strings[j])) {
					hasPrefix = true;
					break;
				}
			}
		}
	}
}

File: go/trie/random_test_data.go
package trie

import (
	"math/rand/v2"
	"strings"
)

type randomTestData struct {
	strings   []string
	hasPrefix bool
}

var random = rand.New(rand.NewPCG(0, 0))

func newRandomTestData() *randomTestData {
	td := &randomTestData{}

	n := 1 + random.IntN(26)
	m := 1 + random.IntN(20)

	td.strings = make([]string, 1+random.IntN(20))
	for i := range td.strings {
		buf := make([]byte, 1+random.IntN(m))
		for j := range buf {
			buf[j] = byte('a' + random.IntN(n))
		}
		td.strings[i] = string(buf)
	}

	for i := range td.strings {
		for j := range td.strings {
			if i != j && strings.HasPrefix(td.strings[i], td.strings[j]) {
				td.hasPrefix = true
				break
			}
		}
	}

	return td
}
